src/stc/PlatWX.cpp

```diff
@@ -1983,433 +1997,743 @@ PRectangle Window::GetMonitorRect(Point pt) {
 
-void ListBoxImpl::SetFont(Font &font) {
-    GETLB(wid)->SetFont(*((wxFont*)font.GetID()));
-}
+private:
+    WX_DECLARE_HASH_SET(int, wxIntegerHash, wxIntegerEqual, SetOfInts);
+
+    wxSTCListBoxVisualData* m_visualData;
+    wxVector<wxString>      m_labels;
+    wxVector<int>           m_imageNos;
+    size_t                  m_maxStrWidth;
+    int                     m_currentRow;
+
+    CallBackAction          m_doubleClickAction;
+    void*                   m_doubleClickActionData;
+    int                     m_aveCharWidth;
+
+    // These drawing parameters are computed or set externally.
+    int m_borderSize;
+    int m_textHeight;
+    int m_itemHeight;
+    int m_textTopGap;
+    int m_imageAreaWidth;
+    int m_imageAreaHeight;
+
+    // These drawing parameters are set internally and can be changed if needed
+    // to better match the appearance of a list box on a specific platform.
+    int m_imagePadding;
+    int m_textBoxToTextGap;
+    int m_textExtraVerticalPadding;
+};
 
+wxSTCListBox::wxSTCListBox(wxWindow* parent, wxSTCListBoxVisualData* v, int ht)
+             :wxSystemThemedControl<wxVListBox>(),
+              m_visualData(v), m_maxStrWidth(0), m_currentRow(wxNOT_FOUND),
+              m_doubleClickAction(NULL), m_doubleClickActionData(NULL),
+              m_aveCharWidth(8), m_textHeight(ht), m_itemHeight(ht),
+              m_textTopGap(0), m_imageAreaWidth(0), m_imageAreaHeight(0)
+{
+    wxVListBox::Create(parent, wxID_ANY, wxDefaultPosition, wxDefaultSize,
+                       wxBORDER_NONE);
+
+    m_imagePadding             = FromDIP(1);
+    m_textBoxToTextGap         = FromDIP(3);
+    m_textExtraVerticalPadding = FromDIP(1);
+
+    SetBackgroundColour(m_visualData->GetBgColour());
+
+    Bind(wxEVT_LISTBOX_DCLICK, &wxSTCListBox::OnDClick, this);
+    Bind(wxEVT_SYS_COLOUR_CHANGED, &wxSTCListBox::OnSysColourChanged, this);
 
-void ListBoxImpl::Create(Window &parent, int ctrlID, Point location_, int lineHeight_, bool unicodeMode_, int WXUNUSED(technology_)) {
-    location = location_;
-    lineHeight =  lineHeight_;
-    unicodeMode = unicodeMode_;
-    maxStrWidth = 0;
-    wid = new wxSTCListBoxWin(GETWIN(parent.GetID()), ctrlID, location);
-    if (imgList != NULL)
-        GETLB(wid)->SetImageList(imgList, wxIMAGE_LIST_SMALL);
+    if ( m_visualData->HasListCtrlAppearance() )
+    {
+        EnableSystemTheme();
+        Bind(wxEVT_MOTION, &wxSTCListBox::OnMouseMotion, this);
+        Bind(wxEVT_LEAVE_WINDOW, &wxSTCListBox::OnMouseLeaveWindow, this);
+
+        #ifdef __WXMSW__
+            // On MSW when using wxRendererNative to draw items in list control
+            // style, the colours used seem to be based on the parent's
+            // background colour. So set the popup's background.
+            parent->SetOwnBackgroundColour(m_visualData->GetBgColour());
+        #endif
+    }
 }
 
+wxSTCListBox::~wxSTCListBox()
+{
+    m_labels.clear();
```

Comment:
This is not really needed, why manually do something done automatically anyhow?

Reply:
This has been removed.